### PR TITLE
[fix](jdbc) fix jdbc driver bug and external datasource p2 test case issue

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. Fix bug that when create jdbc resource with only jdbc driver file name, it will failed to do checksum
    This is because we forgot the pass the full driver url to JdbcClient.

2. Add `ResultSet.FETCH_FORWARD` and set AutoCommit to false to jdbc connection, so to avoid OOM when fetching large amount of data

3. set `useCursorFetch` in jdbc url for both MySQL and PostgreSQL.

4. Fix some p2 external datasource bug

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
5. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
6. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
7. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
8. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.56 seconds
 load time: 522 seconds
 storage size: 17121900929 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230117200542_clickbench_pr_82622.html

File: fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java [hunk 1]
```diff
@@ -248,9 +248,11 @@ private void init(String driverUrl, String sql, int batchSize, String driverClas
 
             dataSource = new HikariDataSource(config);
             conn = dataSource.getConnection();
+            conn.setAutoCommit(false);
             if (op == TJdbcOperation.READ) {
```
Comment (zhangstar333) [inline fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:252]: should set it to false only the op==READ

Comment (morningman) [reply fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:251]: done

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**